Can't use any plugins

Since #1507 was merged, any built wallet app can't load plugins.

If you run `yarn dev` it works fine but if you try building the wallet with `yarn build` and execute the app, all plugins error out with a message:
`Could not enable '<plugin name>' plugin for profile '<profile name>': Cannot find module 'vue/dist/vue.common.js'` 

This happens with my own locally built wallet app and also any of the published artifacts on GitHub from #1507's merge onwards.
 
**Expected Behavior**

Plugins should work

**Current Behavior**

Plugins don't work

**Possible Solution**

Make plugins work

**Steps to Reproduce (for bugs)**

Build the wallet and run it (i.e. use `yarn build` not `yarn dev`). Install any plugin and enable it. The plugins icon in the menu won't appear and the console log will show the error above.

**Your Environment**

- Version used: `develop` branch
- Operating System and version: macOS Catalina 10.15.1

--- a/__tests__/unit/services/plugin-manager/plugin-sandbox.spec.js
+++ b/__tests__/unit/services/plugin-manager/plugin-sandbox.spec.js
@@ -1,0 +1,63 @@
+import path from 'path'
+import { createLocalVue } from '@vue/test-utils'
+import { PluginSandbox } from '@/services/plugin-manager/plugin-sandbox'
+import { Plugin } from '@/services/plugin-manager/plugin'
+
+const rootPath = path.resolve(__dirname, '../../../')
+const app = createLocalVue()
+const plugin = new Plugin({
+  rootPath,
+  fullPath: path.resolve(__dirname),
+  config: {
+    permissions: []
+  }
+})
+
+beforeEach(() => {
+  plugin.config.permissions = []
+})
+
+describe('Plugin Sandbox', () => {
+  it('should parse file with api', async () => {
+    const sandbox = new PluginSandbox({
+      app,
+      plugin
+    })
+
+    await sandbox.install()
+
+    const pluginVM = sandbox.getComponentVM()
+    expect(pluginVM.run(`
+      module.exports = walletApi
+    `)).toBeDefined()
+  })
+
+  it('should parse file without api', async () => {
+    const sandbox = new PluginSandbox({
+      app,
+      plugin
+    })
+
+    await sandbox.install()
+
+    const pluginVM = sandbox.getPluginVM()
+    expect(() => pluginVM.run(`
+      module.exports = walletApi
+    `)).toThrowError('walletApi is not defined')
+  })
+
+  it('should read permissions', async () => {
+    plugin.config.permissions = ['ALERTS']
+    const sandbox = new PluginSandbox({
+      app,
+      plugin
+    })
+
+    await sandbox.install()
+
+    const componentVM = sandbox.getComponentVM()
+    expect(() => componentVM.run(`
+      module.exports = walletApi.alert
+    `)).toBeDefined()
+  })
+})

--- a/__tests__/unit/services/plugin-manager/setup/components-setup.spec.js
+++ b/__tests__/unit/services/plugin-manager/setup/components-setup.spec.js
@@ -31,7 +31,10 @@ const pluginObject = {
 }
 
 const sandbox = {
-  getVM: jest.fn(() => ({
+  getComponentVM: jest.fn(() => ({
+    run: jest.fn()
+  })),
+  getPluginVM: jest.fn(() => ({
     run: jest.fn()
   })),
   app: {}

--- a/src/renderer/services/plugin-manager/plugin-sandbox.js
+++ b/src/renderer/services/plugin-manager/plugin-sandbox.js
@@ -40,7 +40,6 @@ export class PluginSandbox {
 
     this.sandbox = {}
     this.walletApi = {}
-
     this.sandboxes = this.__mapPermissionsToSandbox()
   }
 
@@ -57,28 +56,34 @@ export class PluginSandbox {
     }
   }
 
-  getVM ({ loadApi }) {
-    const fullPath = this.plugin.fullPath
-
+  getComponentVM () {
     return new NodeVM({
-      sandbox: this.getSandbox(loadApi),
+      sandbox: this.getSandbox(true),
       require: {
         builtin: [],
         context: 'sandbox',
-        resolve: function (source) {
-          return path.resolve(fullPath, source)
-        },
         external: {
-          modules: [
-            path.resolve(fullPath),
-            'vue/dist/vue.common.js'
-          ],
+          modules: [],
           transitive: true
         },
         root: [
-          this.plugin.rootPath,
-          path.resolve(fullPath)
+          path.resolve(this.plugin.fullPath)
         ]
+      }
+    })
+  }
+
+  getPluginVM () {
+    const { rootPath } = this.plugin
+
+    return new NodeVM({
+      sandbox: this.getSandbox(false),
+      require: {
+        context: 'sandbox',
+        resolve: function (source) {
+          return path.resolve(rootPath, 'node_modules', source)
+        },
+        external: ['vue']
       }
     })
   }

--- a/src/renderer/services/plugin-manager/plugin-setup.js
+++ b/src/renderer/services/plugin-manager/plugin-setup.js
@@ -38,7 +38,7 @@ export class PluginSetup {
     localVue.options._base = localVue
     this.vue = localVue
 
-    this.pluginObject = this.sandbox.getVM(false).run(
+    this.pluginObject = this.sandbox.getPluginVM().run(
       fs.readFileSync(path.join(plugin.fullPath, 'src/index.js')),
       path.join(plugin.fullPath, 'src/index.js')
     )

--- a/src/renderer/services/plugin-manager/setup/components-setup.js
+++ b/src/renderer/services/plugin-manager/setup/components-setup.js
@@ -23,8 +23,8 @@ export function createComponentsSetup (plugin, pluginObject, sandbox, vue) {
           vue,
           fullPath,
           name: componentName,
-          pluginVM: sandbox.getVM({ loadApi: false }),
-          componentVM: sandbox.getVM({ loadApi: true }),
+          pluginVM: sandbox.getPluginVM(),
+          componentVM: sandbox.getComponentVM(),
           logger: sandbox.app.$logger
         })
 

